### PR TITLE
[CMD-714] Python script to clear errant enums.

### PR DESCRIPTION
**Pull request by ChandraAddala**

### CODE REVIEW

**Comment by usnavi** [comment]
lgtm - but i've never worked with python.  Two questions:
1) I assume the blank ``contrib/enum_metrics/enum_metrics/__init__.py`` is required?
2) Is there any automated mechanism which runs the unit tests, say in the PR-builder or elsewhere?

**Comment by ChandraAddala** [comment]
@usnavi 
1) Thats correct. Having __init__.py makes that a package. 
2) No. I have to look into that.

File: contrib/enum_metrics/enum_metrics/clear_errant_enums.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import sys
+import argparse
+import json
+import dbclient as db
+import esclient as es
```
Comment (GeorgeJahad) [inline contrib/enum_metrics/enum_metrics/clear_errant_enums.py]: i don't see the esclient anywhere.  Am I missing something?

File: contrib/enum_metrics/enum_metrics/dbclient.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        database results.
+
+        :param results:
+        :return:
+        """
+
+        return [(x.key, x.column1, x.value) for x in results]
+
+
+    def get_metrics_excess_enums(self, metric_name):
```
Comment (GeorgeJahad) [inline contrib/enum_metrics/enum_metrics/dbclient.py]: it looks like the get and delete methods below are all identical except for the column family.  Wouldn't it be better to add the CF as a parameter and create a single get and delete method? Within reason, I like DRY code.

File: contrib/enum_metrics/test/test_clear_errant_enums.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+from unittest import TestCase
```
Comment (GeorgeJahad) [inline contrib/enum_metrics/test/test_clear_errant_enums.py]: would it be hard to use something like this:
https://pypi.python.org/pypi/testing.cassandra
to create a dummy cassandra server and dummy data, to create some more extensive unit tests?

File: contrib/enum_metrics/enum_metrics/clear_errant_enums.py (hunk 1 above)
Comment (ChandraAddala) [inline contrib/enum_metrics/enum_metrics/clear_errant_enums.py]: Missed adding that file...Its in the PR now..

File: contrib/enum_metrics/enum_metrics/dbclient.py (hunk 2 above)
Comment (ChandraAddala) [inline contrib/enum_metrics/enum_metrics/dbclient.py]: I thought about it when writing the code. It felt like I am writing dynamic SQL. I thought I would rather have simple select queries for each table. Since we are not doing any complex query, may be I could extract that logic to one method.

File: contrib/enum_metrics/test/test_clear_errant_enums.py (hunk 3 above)
Comment (ChandraAddala) [inline contrib/enum_metrics/test/test_clear_errant_enums.py]: I am exploring that currently. I thought I will create this PR, get everyone's opinion to see if I am going in the right direction. If it works out okay, I will submit it in my subsequent PR. I am also looking at somehting similar for elastic search.

File: contrib/enum_metrics/enum_metrics/dbclient.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        """
+
+        key = tenant_id + '.' + metric_name
+        return self.__get_metrics(key, "metrics_excess_enums")
+
+    def get_metrics_preaggregated_full(self, key):
+        """
+        Retrieve data from metrics_preaggregated_full
+
+        :param key:
```
Comment (GeorgeJahad) [inline contrib/enum_metrics/enum_metrics/dbclient.py]: I would prefer you either complete the param/return comments, or remove them.  To have them half done like this is distracting.

File: contrib/enum_metrics/config.ini [hunk 5]
```diff
@@ -0,0 +1,19 @@
+[localhost]
+cassandra_nodes = localhost
+es_nodes = localhost
+
```
Comment (GeorgeJahad) [inline contrib/enum_metrics/config.ini]: This file seems rax specific.  We try not to check such files in to the open BF repo.

Instead this may be a separate PR for bf-chef.  Pull all but localhost out for now.

**Comment by GeorgeJahad** [comment]
+1 after removing ip addrs.

nice work!

**Comment by ChandraAddala** [comment]
Removed the IP addresses of other environments.

File: contrib/enum_metrics/README.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
+enum_metrics python utilities
+=============================
+
+This project contains some python utilities to work with enum metrics. 
+
+### Setup
+
+Get the [blue flood](https://github.com/rackerlabs/blueflood) repo from github. Execute the following commands
```
Comment (izrik) [inline contrib/enum_metrics/README.md]: Typo: should be `blueflood`, one word.

File: contrib/enum_metrics/enum_metrics/errant_enums.py [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+    delete_parser = subparsers.add_parser('delete', help='Delete errant enum')
+
+    delete_parser.add_argument('--dryrun',
+                               action='store_true', help='Display errant enums related data for the given metric name.')
+    delete_parser.add_argument('-m', '--metricName',
+                               required=True, help='metric name to be deleted')
+    delete_parser.add_argument('-t', '--tenantId',
+                               required=True, help='tenantId corresponding to the metric name to be deleted')
+
+    environments = [LOCALHOST, QE01, QE02, STAGING, PROD]
```
Comment (izrik) [inline contrib/enum_metrics/enum_metrics/errant_enums.py]: Only one of these environments is present in the `config.ini` file. It would be better to use the 'SafeConfigParser.sections()` method to get the list of available environments, instead of hard-coding them. Also, these choices are rax-specific.

**Comment by izrik** [comment]
Just out of curiosity, why is this in python, whereas the rest of the project is in java?

File: contrib/enum_metrics/enum_metrics/errant_enums.py [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+import dbclient as db
+import esclient as es
+import config as cf
+
+LOCALHOST = 'localhost'
+QE01 = 'qe01'
+QE02 = 'qe02'
+STAGING = 'staging'
+PROD = 'prod'
+
```
Comment (drenalin23) [inline contrib/enum_metrics/enum_metrics/errant_enums.py]: Staging = stage
Don't forget perf01.

File: contrib/enum_metrics/enum_metrics/errant_enums.py (hunk 7 above)
Comment (ChandraAddala) [inline contrib/enum_metrics/enum_metrics/errant_enums.py]: I addressed this in the PR https://github.com/rackerlabs/blueflood/pull/575

File: contrib/enum_metrics/enum_metrics/errant_enums.py (hunk 8 above)
Comment (ChandraAddala) [inline contrib/enum_metrics/enum_metrics/errant_enums.py]: @drenalin23  I just made the script environment agnostic. It basically works with whatever environment we configure in config.ini.

**Comment by ChandraAddala** [comment]
@izrik I started out as a command line script to clean up some data. Python made lot of sense. This project seems to host all these python scripts under contrib folder. Thats why I placed the script over there.

**Comment by izrik** [comment]
OK. I'm all for python. It's easy to program in. Nevertheless, I find multi-language projects are a little more work overall. Is this script only to be executed on-demand, or does it need to be integrated into our CI/CD system?

**Comment by ChandraAddala** [comment]
It will be executed on-demand.

**Comment by GeorgeJahad** [comment]
FYI @izrik, there are two other critical pieces of the project that are in python, in the blueflood directory,, contrib/grinder, and contrib/graphite
